sound/soc/sof/ipc4-topology.c

```diff
@@ -1589,8 +1590,48 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 		return -EINVAL;
 	}
 
+	/* check if all input formats are the same */
+	single_input_format = sof_ipc4_copier_is_single_format(sdev,
+								available_fmt->input_pin_fmts,
+								available_fmt->num_input_formats);
+
+	/*
+	 * in the case of capture, if all the input formats are the same, update the reference
```

Comment:
So, if all the input formats are the same then why we have multiple of them? Same format, but different rate?